core/src/main/java/org/apache/carbondata/core/datastore/page/UnsafeFixLengthColumnPage.java

```diff
@@ -359,38 +412,36 @@ public void freeMemory() {
     }
   }
 
-  @Override
-  public void convertValue(ColumnPageValueConverter codec) {
-    int pageSize = getPageSize();
+  @Override public void convertValue(ColumnPageValueConverter codec) {
     if (dataType == DataTypes.BYTE) {
-      for (long i = 0; i < pageSize; i++) {
+      for (long i = 0; i < totalLength / ByteUtil.SIZEOF_BYTE; i++) {
```

Comment:
for loop end condition (totalLength / ByteUtil.SIZEOF_BYTE) is evaluated for every row.  if we extract the computation of page size to a method, we can avoid this

Reply:
Done